typescript.js

```diff
@@ -9,15 +9,19 @@ module.exports = {
       parserOptions: { warnOnUnsupportedTypeScriptVersion: false },
       extends: ['plugin:@typescript-eslint/recommended'],
       rules: {
-        'no-use-before-define': 'off', // Turning off because we have the TS version off
-        '@typescript-eslint/no-use-before-define': 'off',
-        '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
-
+        'no-use-before-define': 'off', // @typescript-eslint/no-use-before-defined requires this to be off
+        '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
         'no-unused-vars': 'off', // @typescript-eslint/no-unused-vars requires this to be off
-        '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
-
-        // TS complains about using the .ts/.tsx extensions in the import statement. so turning this rule off for TS
-        'import/extensions': 'off',
+        'no-empty-functions': 'off', // @typescript-eslint/no-empty-function requires this to be off
+        '@typescript-eslint/no-empty-function': ['error', { allow: ['arrowFunctions'] }],
+        '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
+      },
+    },
+    {
+      files: ['*.test.ts?(x)'],
+      rules: {
+        '@typescript-eslint/no-non-null-assertion': 'off', // We can know for sure if something is null or not for tests because we made the mock data
```

Comment:
Note from Jordi:
@typescript-eslint/no-non-null-assertion - turns off in test files because we can know for sure if something is null or not for tests because we made the mock data
Example:
![image](https://user-images.githubusercontent.com/3885959/215878495-2ed28c81-a8aa-42ca-9cb0-6f5353c2f87f.png)
vs
![image](https://user-images.githubusercontent.com/3885959/215878545-c25ebed2-8206-45d3-9bcd-456fd9fa031a.png)